### ISSUE TITLE
LNM own flight plan format

### ISSUE THREAD

**Issue by albar965** [albar965/littlenavmap#293]
Create own XML and PLN similar flight plan format which avoids limitations of annotated PLN.

Save and load:
* Metadata: Program version, creation date, AIRAC cycle, etc., database version
* Description text for whole flight plan.
* Description for waypoints.
* All procedure information
* Userpoint information including name, ident and description
* Aircraft performance (probably), climb, ToD and more.
* Redundancy for waypoints: Store name, ICAO, region, position (decimal degrees) and navaid type

Keep annotated old PLN format but make LNM default.

Use Flight Gear format since it has all needed features:
[Route manager flightplans](http://wiki.flightgear.org/Flightplan_XML_formats#Route_manager_flightplans)

**Comment by Gulliver54** [comment]
Hi Alex,

Very useful I think.

About aircraft performance:
Do you plan to create a aircraft database so users can add aircraft and performance LNM can use?
If so, is it possible to exchange those files with other users?

Regards,
Gradus

**Comment by albar965** [comment]
First I'd like to keep it very simple like #255 and later on implement #38 which might even be an additional program.
I'll use XML files (users can edit manually if they wish) and one file per aircraft so exchanging is easy.
Alex

**Comment by albar965** [comment]
Moved to 2.6. This is better implemented with #165.

**Comment by albar965** [comment]
Done. Example:

``` XML
<?xml version="1.0" encoding="UTF-8"?>
<LittleNavmap>
  <Flightplan>
    <Header>
      <FlightplanType>IFR</FlightplanType>
      <CruisingAlt>30000</CruisingAlt>
      <CreationDate>2020-05-03T22:18:43</CreationDate>
      <FileVersion>0.9</FileVersion>
      <ProgramName>Little Navmap</ProgramName>
      <ProgramVersion>2.5.0.develop</ProgramVersion>
    </Header>
    <SimData Cycle="1802">XP11</SimData>
    <NavData Cycle="2005">NAVIGRAPH</NavData>
    <AircraftPerformance>
      <FilePath>/home/alex/Downloads/FJS 722v3.lnmperf</FilePath>
      <Type>B722</Type>
      <Name>FJS 722v3</Name>
    </AircraftPerformance>
    <Departure>
      <Pos Lon="-120.540169" Lat="46.569576" Alt="1099.00"/>
      <Start>Gate 2</Start>
    </Departure>
    <Procedures>
      <SID>
        <Name>WENAS7</Name>
        <Runway>09</Runway>
        <Transition>PERTT</Transition>
      </SID>
      <STAR>
        <Name>PIGLU4</Name>
        <Runway>16</Runway>
        <Transition>YDC</Transition>
      </STAR>
      <Approach>
        <Name>TATVI</Name>
        <ARINC>I16-Z</ARINC>
        <Runway>16</Runway>
        <Type>ILS</Type>
        <Suffix>Z</Suffix>
        <Transition>HUMEK</Transition>
        <TransitionType>F</TransitionType>
      </Approach>
    </Procedures>
    <Alternates>
      <Alternate>
        <Name>Penticton</Name>
        <Ident>CYYF</Ident>
        <Type>AIRPORT</Type>
        <Pos Lon="-119.602280" Lat="49.462578" Alt="1128.00"/>
      </Alternate>
      <Alternate>
        <Name>Grand Forks</Name>
        <Ident>CZGF</Ident>
        <Type>AIRPORT</Type>
        <Pos Lon="-118.430466" Lat="49.015629" Alt="1409.00"/>
      </Alternate>
    </Alternates>
    <Waypoints>
      <Waypoint>
        <Name>Yakima Air / Mcallister Field</Name>
        <Ident>KYKM</Ident>
        <Type>AIRPORT</Type>
        <Pos Lon="-120.543999" Lat="46.568199" Alt="1099.00"/>
      </Waypoint>
      <Waypoint>
        <Ident>COBDI</Ident>
        <Region>K1</Region>
        <Type>WAYPOINT</Type>
        <Pos Lon="-120.662415" Lat="47.119732" Alt="15168.10"/>
      </Waypoint>
      <Waypoint>
        <Name>User defined point</Name>
        <Ident>USERPT</Ident>
        <Region>K1</Region>
        <Type>USER</Type>
        <Pos Lon="-120.848000" Lat="47.676601" Alt="25819.59"/>
      </Waypoint>
      <Waypoint>
        <Ident>ROZSE</Ident>
        <Region>K1</Region>
        <Type>WAYPOINT</Type>
        <Pos Lon="-121.029366" Lat="48.222935" Alt="30000.00"/>
      </Waypoint>
      <Waypoint>
        <Ident>DIABO</Ident>
        <Region>K1</Region>
        <Type>WAYPOINT</Type>
        <Pos Lon="-120.937080" Lat="48.833759" Alt="30000.00"/>
      </Waypoint>
      <Waypoint>
        <Ident>IWACK</Ident>
        <Region>K1</Region>
        <Airway>J503</Airway>
        <Type>WAYPOINT</Type>
        <Pos Lon="-120.837067" Lat="48.932140" Alt="30000.00"/>
      </Waypoint>
      <Waypoint>
        <Ident>CFKNF</Ident>
        <Region>K1</Region>
        <Airway>J503</Airway>
        <Type>WAYPOINT</Type>
        <Pos Lon="-120.767761" Lat="49.000000" Alt="30000.00"/>
      </Waypoint>
      <Waypoint>
        <Ident>FOLDY</Ident>
        <Region>CY</Region>
        <Airway>J503</Airway>
        <Type>WAYPOINT</Type>
        <Pos Lon="-120.711891" Lat="49.051582" Alt="30000.00"/>
      </Waypoint>
      <Waypoint>
        <Name>Kelowna</Name>
        <Ident>CYLW</Ident>
        <Type>AIRPORT</Type>
        <Pos Lon="-119.377998" Lat="49.957199" Alt="1461.00"/>
      </Waypoint>
    </Waypoints>
  </Flightplan>
</LittleNavmap>
```
